reconcile/utils/saasherder.py

```diff
@@ -387,48 +406,58 @@ def _process_template(self, options):
                 logging.error(
                     f"[{url}/{path}:{target_ref}] " +
                     f"error fetching template: {str(e)}")
-                return None, None
-
-            if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                return None, None, None
+
+            # add IMAGE_TAG only if it is required and unspecified
+            if self._parameter_required_and_absent(
+                    "IMAGE_TAG", consolidated_parameters, template):
+                sha_substring = commit_sha[:hash_length]
+                # IMAGE_TAG takes one of two forms:
+                # - If saas file attribute 'use_channel_in_image_tag' is true,
+                #   it is {CHANNEL}-{SHA}
+                # - Otherwise it is just {SHA}
+                if self._get_saas_file_attribute("use_channel_in_image_tag"):
```

Comment:
You'll need to add this field to the graphql query and set it in the saasherder init I believe?
It would also require a schema change in app-interface to support this change.

Reply:
Found the query thing. Will look for the schema.

Reply:
https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/17433 for the schema and documentation change. I suppose that should be merged with/after promoting this PR.

Reply:
:heart: